### PR TITLE
Add a testnet similar to sc_network_test.

### PR DESCRIPTION
**Pull request by expenses**
This PR ports `substrate-test-runtime-client` and `sc-network-test` over to Polkadot.

See https://github.com/paritytech/polkadot/issues/833

### CODE REVIEW

File: network/test/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,901 @@
+// Copyright 2017-2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
```
Comment (rphmeier) [inline network/test/src/lib.rs]: Should replace occurrences of `Substrate` with `Polkadot`.

File: network/test/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,901 @@
+// Copyright 2017-2020 Parity Technologies (UK) Ltd.
```
Comment (rphmeier) [inline network/test/src/lib.rs]: Copyright notice years are probably wrong.

File: network/test/src/sync.rs [hunk 3]
```diff
@@ -0,0 +1,662 @@
+// Copyright 2017-2020 Parity Technologies (UK) Ltd.
```
Comment (rphmeier) [inline network/test/src/sync.rs]: This file is unneeded

Comment (rphmeier) [reply network/test/src/sync.rs]: (this functionality is tested in Substrate - Polkadot doesn't change the sync code)

File: runtime/test-runtime/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,848 @@
...
+		Authorship: authorship::{Module, Call, Storage},
+		Staking: staking,
+		Offences: offences::{Module, Call, Storage, Event},
+		Session: session::{Module, Call, Storage, Event, Config<T>},
+		FinalityTracker: finality_tracker::{Module, Call, Inherent},
+		Grandpa: grandpa::{Module, Call, Storage, Config, Event},
+		ImOnline: im_online::{Module, Call, Storage, Event<T>, ValidateUnsigned, Config<T>},
+		AuthorityDiscovery: authority_discovery::{Module, Call, Config},
+
+		// Governance stuff; uncallable initially.
```
Comment (rphmeier) [inline runtime/test-runtime/src/lib.rs]: I'm not convinced we need the governance pallets for testing Polkadot functionality.

File: runtime/test-runtime/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,848 @@
...
+		Registrar: registrar::{Module, Call, Storage, Event, Config<T>},
+
+		// Utility module.
+		Utility: utility::{Module, Call, Storage, Event<T>},
+
+		// Less simple identity module.
+		Identity: identity::{Module, Call, Storage, Event<T>},
+
+		// Society module.
+		Society: society::{Module, Call, Storage, Event<T>},
```
Comment (rphmeier) [inline runtime/test-runtime/src/lib.rs]: Likewise with Society, Identity, Recovery, and possibly Utility

File: runtime/test-runtime/src/lib.rs [hunk 6]
```diff
@@ -0,0 +1,848 @@
...
+		TechnicalCommittee: collective::<Instance2>::{Module, Call, Storage, Origin<T>, Event<T>, Config<T>},
+		ElectionsPhragmen: elections_phragmen::{Module, Call, Storage, Event<T>},
+		TechnicalMembership: membership::<Instance1>::{Module, Call, Storage, Event<T>, Config<T>},
+		Treasury: treasury::{Module, Call, Storage, Event<T>},
+
+		// Claims. Usable initially.
+		Claims: claims::{Module, Call, Storage, Event<T>, Config<T>, ValidateUnsigned},
+
+		// Parachains stuff; slots are disabled (no auctions initially). The rest are safe as they
+		// have no public dispatchables.
```
Comment (rphmeier) [inline runtime/test-runtime/src/lib.rs]: These are the core modules that we're actually going to want to test.

**Comment by rphmeier** [comment]
needs a merge w/ master, then should be good to go.

**Comment by expenses** [comment]
This should be ready now.

**Comment by expenses** [comment]
@rphmeier what should I be doing to fix the gitlab-check-runtime test?

**Comment by rphmeier** [comment]
@expenses You should bump the `spec_version` and/or `impl_version` in the runtimes (according to the rules 

"If changes made do not alter logic, just bump 'impl_version'. If they do change logic, bump 'spec_version'."

**Pull request merged**